Tools/Scripts/webkitpy/__init__.py

```diff
@@ -118,3 +133,77 @@
 AutoInstall.register(Package('webkitbugspy', Version(0, 3, 1)), local=True)
 
 import webkitscmpy
+
+# Disable IPV6 if Bugzilla can't use IPV6.
+import six
+import socket
+import urllib3
+
+HAS_CHECKED_IPV6 = False
+FIRST_IPV6_TIMEOUT = 3
+
+
+# Pulled from urllib3, https://github.com/urllib3/urllib3/blob/main/src/urllib3/util/connection.py
+# Disable IPV6 if the first attempt to connect to an IPV6 address fails
+def create_connection(
+    address,
+    timeout=socket._GLOBAL_DEFAULT_TIMEOUT,
+    source_address=None,
+    socket_options=None,
+):
+    global HAS_CHECKED_IPV6
+
+    host, port = address
+    if host.startswith("["):
+        host = host.strip("[]")
+    err = None
+
+    family = urllib3.util.connection.allowed_gai_family()
+
+    try:
+        host.encode("idna")
+    except UnicodeError:
+        return six.raise_from(
+            LocationParseError(u"'%s', label empty or too long" % host), None
+        )
+
+    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
+        af, socktype, proto, canonname, sa = res
+
+        is_first_ipv6 = not HAS_CHECKED_IPV6 and af == socket.AF_INET6
+        if af == socket.AF_INET6:
+            if not urllib3.util.connection.HAS_IPV6:
+                continue
+            HAS_CHECKED_IPV6 = True
```

Comment:
Nit: I feel we set this to True prematurely based on the variable name. Also, is this thread-safe?